src/osd/OSD.cc

```diff
@@ -7678,6 +7690,33 @@ vector<DaemonHealthMetric> OSD::get_health_metrics()
       if (slow) {
 	derr << __func__ << " reporting " << slow << " slow ops, oldest is "
 	     << oldest_op->get_desc() << dendl;
+        if (!log_slowop) {
+          int max_slow = -1;
+          int slow_poolid = -1;
+          std::for_each(pools.begin(), pools.end(),
+                  [&slow_poolid, &max_slow](std::pair<const int, int>& pp) {
+                     if (max_slow < pp.second) {
+                        max_slow = pp.second;
+                        slow_poolid = pp.first;
+                     }
+                  });
+          string pool_name = osdmap->get_pool_name(slow_poolid);
```

Comment:
maybe it's worth asserting `slow_poolid` is not -1/null before getting pool name.

Reply:
assert will cause OSD to crash if poolid is -1/null. Instead check if poolid exists in get_pools() then only print the most affected pool with slow ops count for the pool.